src/components/CatSearch.tsx

```diff
@@ -0,0 +1,247 @@
+                )
+              })
+            }
+          </ul>
+          <button tabIndex={0} onClick={() => toggleAllFilters("experience", true)}>Select All</button>
+          <button tabIndex={0} onClick={() => toggleAllFilters("experience", false)}>Deselect All</button>
+        </details>
+        <details>
+          <summary>Afilliation Filters</summary>
+          <p>(Filters by affiliation to selected cats)</p>
+          <Checkbox label="Enable Filter" checked={affiliationEnabled} onChange={() => setAffiliationEnabled(!affiliationEnabled)}/>
+          <ul>
+            <Checkbox label="Mentors of selected" checked={filters["affiliation"]["apprentice"]} onChange={() => toggleFilter("affiliation", "apprentice")}/>
+            <Checkbox label="Apprentices of selected" checked={filters["affiliation"]["mentor"]} onChange={() => toggleFilter("affiliation", "mentor")}/>
+            <Checkbox label="Mates of selected" checked={filters["affiliation"]["mates"]} onChange={() => toggleFilter("affiliation", "mates")}/>
+          </ul>
+        </details>
+      </div>
+      <div className="cat-search-cats">
+        <div className="cats-list">
+          {
+            catPages[currentPage].filter(checkFilters).map((cat) => {
+              return (
+                <div className="cat cat-search-select">
+                  <Checkbox
+                    label={
+                      <div>
+                        <CatDisplay cat={cat} w="75px" h="75px" />
+                        <div>{cat.name.display}</div>
+                        <div className="cat-search-select-status">{cat.status}</div>
+                      </div>
+                    }
+                    checked={selectedCats.includes(cat.ID)}
+                    onChange={() => toggleCatSelected(cat)}
+                  />
+                </div>
+              );
+            })
+          }
+        </div>
+        <div className="cat-search-page-controls">
+          <button tabIndex={0} 
+            onClick={() => setCurrentPage(Math.max(currentPage - 1, 0))} 
+            disabled={currentPage == 0}
+          >
+            <TbCaretLeftFilled />
+          </button>
+          <div>
+            {currentPage + 1} of {catPages.length}
+          </div>
+          <button tabIndex={0}
+            onClick={() => setCurrentPage(Math.min(currentPage + 1, catPages.length - 1))} 
+            disabled={currentPage == catPages.length - 1}
+          >
+            <TbCaretRightFilled />
+          </button>
+        </div>
+        <div className="cat-search-page-controls">
+          <button tabIndex={0} 
+            onClick={() => selectRandom()} 
+            disabled={selectedCats.length == maxSelection}
+          >
+            Select Random
+          </button>
+          <button tabIndex={0} 
+            onClick={() => selectRandom(maxSelection)}
+          >
+            Select Random All
```

Comment:
```suggestion
            Select Random Group
```
I think this is a little more clear.